layers/+tools/dap/funcs.el

```diff
@@ -0,0 +1,66 @@
+;;; packages.el --- DAP mode functions File for Spacemacs
+;;
+;; Copyright (c) 2012-2018 Sylvain Benner & Contributors
+;;
+;; Author: Ivan Yonchovski (yyoncho@gmail.com)
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+(defun spacemacs/dap-bind-keys-for-mode (mode)
+  "Define key bindings for the specific MODE."
+
+  (spacemacs/declare-prefix-for-mode mode "md" "debug")
+  (spacemacs/declare-prefix-for-mode mode "mdd" "debuging")
+  (spacemacs/declare-prefix-for-mode mode "mdb" "breakpoints")
+  (spacemacs/declare-prefix-for-mode mode "mdl" "debug windows")
+  (spacemacs/declare-prefix-for-mode mode "mdS" "switch")
+  (spacemacs/declare-prefix-for-mode mode "mdI" "inspect")
+  (spacemacs/declare-prefix-for-mode mode "mde" "eval")
+
+  (spacemacs/set-leader-keys-for-major-mode mode
+    ;; debuging/running
+    "ddd" #'dap-debug
+    "ddl" #'dap-debug-last
+    "ddr" #'dap-debug-recent
+    ;; stepping
+    "dc" #'dap-continue
+    "di" #'dap-step-in
+    "do" #'dap-step-out
+    "ds" #'dap-next
+    "dv" #'dap-ui-inspect-thing-at-point
+    "dr" #'dap-restart-frame
+    ;; transient state
+    "d." #'dap-hydra
+    ;; abandon
+    "da" #'dap-disconnect
+    "dA" #'dap-delete-all-sessions
+    ;; eval
+    "dee" #'dap-eval
+    "der" #'dap-eval-region
+    "det" #'dap-eval-thing-at-point
+    ;; switching
+    "dSs" #'dap-switch-session
+    "dSt" #'dap-switch-thread
+    "dSf" #'dap-switch-frame
+    ;; inspect
+    "dIi" #'dap-ui-inspect
+    "dIr" #'dap-ui-inspect-region
+    "dIt" #'dap-ui-inspect-thing-at-point
+    ;; breakpoints
+    "dbb" #'dap-breakpoint-toggle
+    "dbc" #'dap-breakpoint-condition
+    "dbl" #'dap-breakpoint-log-message
+    "dbh" #'dap-breakpoint-hit-condition
+    "dba" #'dap-breakpoint-add
+    "dbd" #'dap-breakpoint-delete
+    "dB"  #'dap-breakpoint-delete-all
+    ;; repl
+    "d'"  #'dap-ui-repl
+    ;; windows
+    "dlo" #'dap-go-to-output-buffer
```

Comment:
Why have you chosen `dl` for this one, since you mention windows I would assume `w` would be more fitting?

Reply:
There is no valid reason - I wanted to avoid as much as possible the standard navigation keys so the hydra could work and also I started with `mdls` which stands for `list session` and then I started using `l` as a prefix for going to the DAP windows.

Reply:
I don't think the hydra should contain toggles for windows anyway and `debug windows` is not a good description for an `l` key. What I suggest is either use `w` for this or change the description to say `layout`, although I would advise more on going with `w`.